lib/_stream_readable.js

```diff
@@ -1070,13 +1074,64 @@ Readable.prototype.wrap = function(stream) {
 };
 
 Readable.prototype[SymbolAsyncIterator] = function() {
+  if (typeof this.read === 'function') {
+    const iter = createAsyncIterator(this);
+    iter.stream = this;
+    return iter;
+  }
+
   if (createReadableStreamAsyncIterator === undefined) {
     createReadableStreamAsyncIterator =
       require('internal/streams/async_iterator');
   }
+
   return createReadableStreamAsyncIterator(this);
 };
 
+async function* createAsyncIterator(stream) {
+  let _resolve = nop;
+  let _reject = nop;
+  let _error = null;
+  let _ended = false;
```

Comment:
What if someone passes an ended stream inside? Is there even a way to know? Do we care?

I think it's fine because a stream shouldn't be ended but not closed and subclasses generally don't violate that (right?).

Reply:
Added a check.